tests/ui/misleading_use_of_ok.rs

```diff
@@ -0,0 +1,17 @@
+#![warn(clippy::misleading_use_of_ok)]
+#![allow(dead_code)]
+
+fn bad_style(x: &str) {
+    x.parse::<u32>().ok();
+}
+
+fn good_style(x: &str) -> Option<u32> {
+    x.parse::<u32>().ok()
+}
+
+#[rustfmt::skip]
+fn strange_parse(x: &str) {
+    x   .   parse::<i32>()   .   ok   ();
+}
+
```

Comment:
Would be good to have some test cases involving macros. Something like
```rs
macro_rules! v {
  () => { Ok::<(), ()>(()) }
}

v!().ok();
```
to exercise the case where only the receiver of the `.ok()` call is from a macro (this sometimes trips up lint suggestions, in my experience), but the current implementation should get this right, given that it's using the syntax context of the `ok()` call expression for the snippet

Reply:
And also one where the whole `stmt.ok();` call is from an external macro, to make sure that checks works correctly. You can take a look at e.g. [this test](https://github.com/rust-lang/rust-clippy/blob/master/tests/ui/unneeded_field_pattern.rs) how it does that. The `external!` macro "tricks" clippy into thinking that it's from an external crate

Reply:
I did implement tests with internal and `external!` macros.
`is_external_macros` seems to not be fooled by the `external!` trick.
I pushed the code anyway, maybe you’ll see my mistake (I barely understand how those things really work).

Reply:
Not entirely sure what's going on there. I'm guessing that it's something to do with the nested `macro_rules!` inside the `external!`. Does it work (i.e. not lint) if you just have something like `external! { Ok::<(), ()>(()).ok(); }`?

Reply:
Yes. `external! { Ok::<(), ()>(()).ok(); }` is not reported by this lint.